### PR TITLE
Implement the remote collectd service

### PR DESCRIPTION
**Pull request by simonpasquier**
This change refactors the collectd formula to be able to install
another collectd instance in charge of running the remote plugins.

### CODE REVIEW

File: collectd/_common.sls [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  pkgrepo.managed:
+  - human_name: Collectd
+  - ppa: nikicat/collectd
+  - file: /etc/apt/sources.list.d/collectd.list
+  - require_in:
+    - pkg: collectd_client_packages
+
+collectd_amqp_packages:
+  pkg.installed:
+  - names:
```
Comment (SwannCroiset) [inline collectd/_common.sls:15]: packages should be defined in map.jinja2 (platform dependent)

File: collectd/remote_client.sls [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+{%- for node_name, node_grains in salt['mine.get']('*', 'grains.items').iteritems() %}
+{%- if node_grains.collectd is defined %}
+{%- set plugins = salt['grains.filter_by']({'default': plugins}, merge=node_grains.collectd.get('remote_plugin', {})) %}
+{%- endif %}
+{%- endfor %}
+
+{%- set client = remote_client %}
+{%- include "collectd/_service.sls" %}
+
+{%- if grains.get('init', None) == 'systemd' %}
```
Comment (SwannCroiset) [inline collectd/remote_client.sls]: remote_client is supported only on xenial/systemd IIUC ?

**Review by SwannCroiset — COMMENTED** [review]
works fine. I've few remarks which can be addressed later or never

File: collectd/remote_client.sls (hunk 2 above)
Comment (elemoine) [reply collectd/remote_client.sls]: I think it's ok to (try to) support both systemd and upstart.

Comment (SwannCroiset) [reply collectd/remote_client.sls]: I do think too, the question would be support it in this this PR or later ;)

File: collectd/_common.sls (hunk 1 above)
Comment (simonpasquier) [reply collectd/_common.sls:15]: This is code that I didn't modified. Also it's only relevant for Ubuntu 10.04 and 12.04 so probably not used anymore.

File: collectd/remote_client.sls (hunk 2 above)
Comment (simonpasquier) [reply collectd/remote_client.sls]: I'll update the PR to support upstart too.

Comment (simonpasquier) [reply collectd/remote_client.sls]: Done

**Review by elemoine — APPROVED** [review]

File: collectd/_common.sls (hunk 1 above)
Comment (SwannCroiset) [reply collectd/_common.sls:15]: ok

**Comment by elemoine** [comment]
@simonpasquier you are probably aware of this already, but collectd on the monitoring node currently sends the metrics to both the metric_collector and the remote_collector (see https://github.com/Mirantis/mk-lab-salt-model/commit/ca19a6f5f2ced9b3d959b401d7d074b1b310ac20). This will need to be changed.

**Comment by simonpasquier** [comment]
@elemoine yes, this is why I need https://github.com/Mirantis/mk-lab-salt-model/pull/74 too.

**Review by SwannCroiset — APPROVED** [review]

**Review by cznewt — APPROVED** [review]
LGTM

**Pull request merged**